### PR TITLE
Updated to support putScomUnderMask

### PR DESCRIPTION
**Pull request by trajeswaran**

### CODE REVIEW

File: src/p10/p10_edbgEcmdDllScom.C [hunk 1]
```diff
@@ -325,7 +325,64 @@ uint32_t p10_dllPutScom(const ecmdChipTarget & i_target, uint64_t i_address, con
...
+
+          const char *path = pdbg_target_path(target);
+          uint64_t xlate_addr = i_address;
+          addr_base = pdbg_address_absolute(target, &xlate_addr);
+
+          if (pdbg_target_probe(target) != PDBG_TARGET_ENABLED)
+              continue;
+
+          // Write the data to the chip if index matches with chip unit number
+          rc = pib_write_mask(target, i_address, i_data.getDoubleWord(0),
```
Comment (RameshIyyar) [inline src/p10/p10_edbgEcmdDllScom.C:368]: is this DLL API used to perform `putscom` on the processor and processor unit?

Looks, we are passing the processor unit target and address to the pib_write_mask, if the ecmd user tried to perform putscom on the processor then, what happens?

File: src/p10/p10_edbgEcmdDllScom.C [hunk 2]
```diff
@@ -325,7 +325,64 @@ uint32_t p10_dllPutScom(const ecmdChipTarget & i_target, uint64_t i_address, con
...
+          return out.error(ECMD_TARGET_NOT_CONFIGURED, FUNCNAME, "Target not configured!\n");
+      }
+
+      // Get the chip level pdbg target for the call to the pib write
+      pdbg_for_each_target(pdbgClassString.c_str(), proc, target) {
+
+          //for "pu" there is no matching required with chip unit number
+          if (i_target.chipUnitType != ""){
+              if (pdbg_target_index(target) != i_target.chipUnitNum)
+                  continue;
```
Comment (RameshIyyar) [inline src/p10/p10_edbgEcmdDllScom.C:357]: By default we are returning the success so, what happens if the user gives some wrong unit index? he will know that giving an unsupported index?

File: src/p10/p10_edbgEcmdDllScom.C [hunk 3]
```diff
@@ -325,7 +325,64 @@ uint32_t p10_dllPutScom(const ecmdChipTarget & i_target, uint64_t i_address, con
...
+          }
+
+          const char *path = pdbg_target_path(target);
+          uint64_t xlate_addr = i_address;
+          addr_base = pdbg_address_absolute(target, &xlate_addr);
+
+          if (pdbg_target_probe(target) != PDBG_TARGET_ENABLED)
```
Comment (RameshIyyar) [inline src/p10/p10_edbgEcmdDllScom.C:364]: By default we are returning the success so, what happens if the identified target is not enabled? The user will know that the given target is not enabled?

Do we not need to fail if the given target is identified to perform putscom but, that's not enabled?

File: src/dll/edbgEcmdDll.C [hunk 4]
```diff
@@ -1352,7 +1352,19 @@ uint32_t dllPutScom(const ecmdChipTarget & i_target, uint64_t i_address, const e
 }
 
 uint32_t dllPutScomUnderMask(const ecmdChipTarget & i_target, uint64_t i_address, const ecmdDataBuffer & i_data, const ecmdDataBuffer & i_mask) {
-  return ECMD_FUNCTION_NOT_SUPPORTED;
+  uint32_t rc = ECMD_SUCCESS;
```
Comment (RameshIyyar) [inline src/dll/edbgEcmdDll.C:1355]: Please add some more details in the commit message like why we are adding this support? is it a new feature in the eBMC or some fix?

File: src/dll/edbgEcmdDll.C [hunk 5]
```diff
@@ -1352,7 +1352,19 @@ uint32_t dllPutScom(const ecmdChipTarget & i_target, uint64_t i_address, const e
...
+  
+  if (pdbg_get_proc() == PDBG_PROC_P10) {
+      rc = p10_dllPutScomUnderMask(i_target,i_address,i_data,i_mask);
+  } else {
+     return ECMD_FUNCTION_NOT_SUPPORTED;
+  }
+
+  if (rc) {
+      return out.error(EDBG_WRITE_ERROR, FUNCNAME, "putscom failed!!");
```
Comment (ojayanth) [inline src/dll/edbgEcmdDll.C:1364]: update message : Putscom under mask failed?

**Review by ojayanth — COMMENTED** [review]
Functionally LGTM.

**Review by thejta — APPROVED** [review]

**Pull request merged**